src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForSRS/OTPAuthorizationForSRSResource.swift

```diff
@@ -22,14 +24,20 @@ struct OTPAuthorizationForSRSResource: Resource {
 		let payload = SAP_Internal_Ppdd_SRSOneTimePassword.with {
 			$0.otp = otpSRS
 		}
+		
 		self.sendResource = ProtobufSendResource(
 			SAP_Internal_Ppdd_SRSOneTimePasswordRequestIOS.with {
-				$0.payload = payload
+				if isFake {
+					// we set the requestPadding only in case of fakeService
+					$0.requestPadding = requestPadding ?? Data()
+				} else {
+					$0.payload = payload
```

Comment:
Looks to me like you're setting a value for `otpSRS`, but it's not used as the payload is not added if `isFake` is true. Not sure what's the right behavior here, if the payload should always be set or the `otpSRS` string is superfluous in the fake scenario

Reply:
@nickguendling I confirmed with Max, it is the expected behaviour `otpSrs is irrelevant in case of fakeSubmission. I would like to still passed the dummy value as I personally would not like to make the type optional for fakeSubmission.